### PR TITLE
Uat backend to develop

### PR DESCRIPTION
**Pull request by prasanna2400**
## Summary by CodeRabbit

- **New Features**
	- Enhanced CI/CD workflows with updated actions for improved performance and security across multiple services.
	- Introduced SARIF report uploads to GitHub for better visibility of dependency vulnerabilities in various components, including `Etreasury`, `ocr-service`, and `sbi-service`.
	- Added a new method for batch retrieval of individual records in the `IndividualService` class, improving functionality for handling multiple UUIDs.

- **Bug Fixes**
	- Corrected branch references in existing workflows to ensure proper triggering and execution based on the new `uat-backend` branch.

- **Documentation**
	- Updated comments and descriptions in workflow files to reflect the changes in branch strategy and deployment focus.

- **Chores**
	- Streamlined CI/CD pipeline configurations across multiple workflows to enhance deployment efficiency and maintainability.
	- Routine maintenance updates to GitHub Actions versions across various workflows for improved reliability and security.

### CODE REVIEW

File: backend/case/src/main/java/org/pucar/dristi/util/CasePdfUtil.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                throw new CustomException(CASE_PDF_UTILITY_EXCEPTION, "Failed to generate PDF, no data received.");
+            }
+
+            log.info("PDF generation successful for case filing Number: {}", caseRequest.getCases().getFilingNumber());
+            return pdfResource;
+        } catch (Exception e) {
+            log.error("Error generating PDF for case filing Number {}", caseRequest.getCases().getFilingNumber(), e);
+            throw new CustomException(CASE_PDF_UTILITY_EXCEPTION, "Error generating PDF for case: " + e.getMessage());
+        }
+    }
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/case/src/main/java/org/pucar/dristi/util/CasePdfUtil.java:47]: @Ajith-beehyv

File: backend/hearing/src/main/java/org/pucar/dristi/util/PdfRequestUtil.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            ResponseEntity<ByteArrayResource> responseEntity = restTemplate.postForEntity(requestUrl.toString(),
+                    requestEntity, ByteArrayResource.class);
+
+            return responseEntity.getBody();
+        } catch (Exception e) {
+            log.error("Error generating PDF for case {}: {}", request, e.getMessage());
+            throw new CustomException(PDF_UTILITY_EXCEPTION, "Error generating PDF for case: " + e.getMessage());
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/hearing/src/main/java/org/pucar/dristi/util/PdfRequestUtil.java:61]: @Ajith-beehyv  @nileshgupta111

File: backend/case/src/main/java/org/pucar/dristi/service/IndividualService.java [hunk 3]
```diff
@@ -47,4 +51,29 @@ public Boolean searchIndividual(RequestInfo requestInfo, String individualId) {
...
+            } else {
+                log.error("No individuals found");
+                return Collections.emptyList();
+            }
+        } catch (Exception e) {
+            log.error("Error in search individual service: ", e);
+            log.error("Individuals not found");
+            return Collections.emptyList();
+        }
+    }
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/case/src/main/java/org/pucar/dristi/service/IndividualService.java:78]: @nileshgupta111

File: backend/hearing/src/main/java/org/pucar/dristi/service/WitnessDepositionPdfService.java [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+        return witnessDepositions;
+    }
+
+    private JsonNode extractAdditionalDetails(Hearing hearing) {
+        return mapper.convertValue(hearing.getAdditionalDetails(), ObjectNode.class);
+    }
+
+    private String getFilingNumber(JsonNode caseDetails) {
+        return caseDetails.has("filingNumber") ? caseDetails.get("filingNumber").asText() : "";
+    }
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/hearing/src/main/java/org/pucar/dristi/service/WitnessDepositionPdfService.java:125]: @Ajith-beehyv

File: backend/hearing/src/main/java/org/pucar/dristi/service/WitnessDepositionPdfService.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+                .caseYear(caseYear)
+                .caseNumber(courtCaseNumber)
+                .name(firstName + " " + lastName)
+                .mobileNumber(mobileNumber)
+                .deposition(deposition)
+                .hearingDate(formatDateFromMillis(hearing.getEndTime()))
+                .taluk(locality)
+                .village(city)
+                .build();
+    }
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/hearing/src/main/java/org/pucar/dristi/service/WitnessDepositionPdfService.java:179]: @Ajith-beehyv

File: backend/hearing/src/test/java/org/pucar/dristi/service/WitnessDepositionPdfServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+        // Use reflection to access private method
+        Method createWitnessObjectsMethod = WitnessDepositionPdfService.class.getDeclaredMethod("createWitnessObjects", RequestInfo.class, Hearing.class);
+        createWitnessObjectsMethod.setAccessible(true);
+
+        // Act
+        Object result = createWitnessObjectsMethod.invoke(service, requestInfo, hearing);
+
+        // Assert
+        assertNotNull(result);
+    }
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/hearing/src/test/java/org/pucar/dristi/service/WitnessDepositionPdfServiceTest.java:150]: @Ajith-beehyv

File: backend/hearing/src/test/java/org/pucar/dristi/service/HearingServiceTest.java [hunk 7]
```diff
@@ -366,4 +366,44 @@ void updateStartAndTime_exception() {
...
+        hearingRequest.setHearing(new Hearing());
+
+        when(validator.validateHearingExistence(requestInfo, hearingRequest.getHearing())).thenReturn(hearingRequest.getHearing());
+
+        // Act
+        Hearing result = hearingService.uploadWitnessDeposition(hearingRequest);
+
+        // Assert
+        assertNotNull(result);
+    }
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/hearing/src/test/java/org/pucar/dristi/service/HearingServiceTest.java:388]: @Ajith-beehyv

File: backend/hearing/src/main/java/org/pucar/dristi/util/CaseUtil.java [hunk 8]
```diff
@@ -46,4 +48,20 @@ public CaseExistsResponse fetchCaseDetails(CaseExistsRequest caseExistsRequest)
...
+        try {
+            response = restTemplate.postForObject(uri.toString(), caseSearchRequest, Map.class);
+            JsonNode jsonNode = mapper.readTree(mapper.writeValueAsString(response));
+            JsonNode caseList = jsonNode.get("criteria").get(0).get("responseList");
+            return caseList.get(0);
+        } catch (Exception e) {
+            log.error(ERROR_WHILE_FETCHING_FROM_CASE, e);
+            throw new CustomException(ERROR_WHILE_FETCHING_FROM_CASE, e.getMessage());
+        }
+    }
```
[the remark this answers is not in the document]
Comment (nileshgupta111) [reply backend/hearing/src/main/java/org/pucar/dristi/util/CaseUtil.java:66]: @prasanna2400  can you assign it to ajith

File: backend/case/src/main/java/org/pucar/dristi/service/CasePdfService.java [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+                            }
+                        }
+                    }
+                }
+            }
+            RequestInfo requestInfo = body.getRequestInfo();
+            CaseRequest caseRequest = CaseRequest.builder().requestInfo(requestInfo).cases(courtCase).build();
+            StringBuilder uri = new StringBuilder(config.getDristiCasePdfHost()).append(config.getDristiCasePdfPath());
+            ByteArrayResource byteArrayResource = casePdfUtil.generateCasePdf(caseRequest, uri);
+            Document document = fileStoreUtil.saveDocumentToFileStore(byteArrayResource.getByteArray(), courtCase.getTenantId());
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/case/src/main/java/org/pucar/dristi/service/CasePdfService.java:77]: @Ajith-beehyv  please do the changes

File: backend/scheduler-svc/src/main/java/digit/service/hearing/OptOutProcessor.java [hunk 10]
```diff
@@ -104,6 +88,30 @@ public void checkAndScheduleHearingForOptOut(HashMap<String, Object> record) {
...
+
+        int totalOptOutCanBeMade = reScheduleHearing.getLitigants().size() + reScheduleHearing.getRepresentatives().size();
+        List<Long> suggestedDates = reScheduleHearing.getSuggestedDates();
+        List<Long> availableDates = reScheduleHearing.getAvailableDates()==null?new ArrayList<>():reScheduleHearing.getAvailableDates();
+        Set<Long> suggestedDatesSet = availableDates.isEmpty() ? new HashSet<>(suggestedDates) : new HashSet<>(availableDates);
+
+        List<OptOut> optOuts = optOutService.search(searchRequest, null, null);
+        int optOutAlreadyMade = optOuts.size();
+        optoutDates.forEach(suggestedDatesSet::remove);
+        reScheduleHearing.setAvailableDates(new ArrayList<>(suggestedDatesSet));
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/scheduler-svc/src/main/java/digit/service/hearing/OptOutProcessor.java:108]: NA

File: backend/case/src/main/resources/application.properties [hunk 11]
```diff
@@ -121,3 +122,25 @@
...
+egov.dristi.case.pdf.path=/dristi-case-pdf/v1/generateCasePdf
+
+egov.localization.search.endpoint=/localization/messages/v1/_search
+
+
+user.oauth.url=https://dristi-kerala-dev.pucar.org/user/oauth/token?_=1713357247536
+
+egov.default.user.username=case-cron-job
+egov.default.user.password=Dristi@123
+egov.user.notification.period=30
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/case/src/main/resources/application.properties:143]: NA

File: backend/case/src/main/java/org/pucar/dristi/config/ServiceConstants.java [hunk 12]
```diff
@@ -111,4 +113,27 @@ private ServiceConstants() {
...
+	public static final String CASE_ADMITTED = "CASE_ADMITTED";
+	public static final String HEARING_REJECTED = "HEARING_REJECTED";
+
+	public static final String HIGH_COURT_LOCALIZATION_CODE = "HIGH_COURT_KERALA";
+	public static final String NOTIFICATION_ENG_LOCALE_CODE = "en_IN";
+	public static final String NOTIFICATION_MODULE_CODE = "notification";
+	public static final String NOTIFICATION_LOCALIZATION_CODES_JSONPATH = "$.messages.*.code";
+
+	public static final String NOTIFICATION_LOCALIZATION_MSGS_JSONPATH = "$.messages.*.message";
+	public static final BigDecimal TAX_AMOUNT = BigDecimal.valueOf(4.00);
```
[the remark this answers is not in the document]
Comment (prasanna2400) [reply backend/case/src/main/java/org/pucar/dristi/config/ServiceConstants.java:132]: NA

**Pull request merged**